LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/IoTHubRegistryManager.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaTools
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Shared;
+
+    public sealed class IoTHubRegistryManager : IDeviceRegistryManager
+    {
+        private readonly RegistryManager instance;
+
+        public static IDeviceRegistryManager From(RegistryManager instance)
```

Comment:
In order to avoid all the CA2000 warning suppressions, you might want to give the task of instantiating the RegistryManager to the IoTHubRegistryManager itself.

As an example, the changes would look like the following

```csharp
public static IDeviceRegistryManager CreateWithProvider(Func<RegistryManager> registryManagerProvider)
{
    return registryManagerProvider == null
        ? throw new ArgumentNullException(nameof(registryManagerProvider))
        : (IDeviceRegistryManager)new IoTHubRegistryManager(registryManagerProvider);
}

private IoTHubRegistryManager(Func<RegistryManager> registryManagerProvider)
{
    this.instance = registryManagerProvider() ?? throw new InvalidOperationException("RegistryManager provider provided a null RegistryManager.");
}
```

and any attempt to instantiate a IoTHubRegistryManager would look like:
```csharp
IoTHubRegistryManager.CreateWithProvider(() => RegistryManager.Create(hostName, new ManagedIdentityCredential()))
```
or
```csharp
IoTHubRegistryManager.CreateWithProvider(() => RegistryManager.CreateFromConnectionString(iotHubConnectionString));
```

@Mandur, wdyt about this alternative approach?